tests/integration-tests/README.md

```diff
@@ -0,0 +1,532 @@
+    │   ├── abilities -> contains the abilities of the actors
+    │   ├── common -> contains the common classes (test constants and helper functions)
+    │   ├── config -> contains the configuration classes (Hoplite)
+    │   ├── features -> contains the features implementation steps
+    │   ├── interactions -> contains the interactions of the actors
+    │   └── runners -> contains the test runners to execute the tests
+    └── resources -> contains the test resources
+        ├── configs -> contains the test configuration files
+        ├── containers -> contains the Docker Compose files to start the test environment
+        └── features -> contains the Gherkin feature files
+```
+
+Here are some rules to follow when writing the tests:
+1. Do not mix steps from different features in the same step definition class. Always create a separate class for each feature.
+2. Do not duplicate configurations, check if the configuration you need already exists in the `test/resources/configs` directory.
+3. Combine scenarios into features based on the functionality they test.
+
+## System under test
+
+The main idea of the framework is to test the OEA as a black box.
+
+The tests interact with the OEA through the API and webhook messages.
+
+<p align="center">
+<img src="docs/static/system_under_test.png" alt="Screenplay pattern overview" width="800"/><br>
+<em>Pic. 2. Overview of the system under test. Roles, Agents and Services communication.</em>
+</p>
+
+### OEA Roles in Tests
+
+| Role     | Description                                           |
+|----------|-------------------------------------------------------|
+| Issuer   | Issues credentials to the holder.                     |
+| Holder   | Stores credentials and presents them to the verifier. |
+| Verifier | Verifies the credentials presented by the holder.     |
+| Admin    | Performs specific administrative tasks                |
+
+- Each OEA can play multiple roles simultaneously after multitenancy is implemented.
+- OEAs can be created on-the-fly or use existing ones, regardless of their origin (local or cloud).
+
+### OEA Configurations
+
+Each OEA can use different configurations for:
+
+| Configuration  | Options                                                               |
+|----------------|-----------------------------------------------------------------------|
+| Secret Storage | Vault, PostgreSQL                                                     |
+| Authentication | ApiKey, JWT Bearer tokens from Keycloak                               |
+| VDR            | PRISM Node in different configs (in-memory, preview/preprod, mainnet) |
+
+The framework allows testing all possible combinations of the above roles and configurations.
+
+## Configuring the tests
+
+Configuration files are located in the `test/resources/configs` directory.
+They are written in [HOCON format](https://github.com/lightbend/config/blob/main/HOCON.md#hocon-human-optimized-config-object-notation).
```

Comment:
They are available in the [HOCON format](https://github.com/lightbend/config/blob/main/HOCON.md#hocon-human-optimized-config-object-notation).